### DISCUSSION TITLE
microdot utemplate in asyncio

### DISCUSSION THREAD

**Discussion by Gingso** [micropython/micropython#9623] [Libraries & Drivers]
Hey,

i tried to use utemplate with microdot on my esp32, with the latest OTA Firmware. But I'm not able to get it running :-/ (microdot itself is running for standard requests)

What i did so far:

- include the Libs for microdot asyncio and utemplate via Thonny on my esp32
- create a folder "templates" which include a "index.html" with some {{ variables }} attributes

implement the following code: <snip>

```
import uasyncio as asyncio
from microdot_asyncio import Microdot, Response
from microdot_utemplate import render_template

app = Microdot()
Response.default_content_type = 'text/html'

..........

@app.route('/')
async def mainpage(request):
    name='myname'
    return render_template(template='index.html', location=name)

```
</snip>

When I connect to the webserver it crashes:

```
Traceback (most recent call last):
  File "/lib/microdot_asyncio.py", line 367, in dispatch_request
  File "/lib/microdot_asyncio.py", line 423, in _invoke_handler
  File "<stdin>", line 130, in mainpage
  File "/lib/microdot_utemplate.py", line 34, in render_template
TypeError: unexpected keyword argument 'location'
```

what i figured out so far:

- in templates an index_html.py file is generated
     --> If i copy paste the included render function to my code and execute directly, everything works fine.

- It looks like an issue with the loader / Load function in microdot_utemplate.py:
```
    if _loader is None:  # pragma: no cover
        init_templates()
    render = _loader.load(template)
    return render(*args, **kwargs)
```

any ideas how to fix this?

_I do not know how this loader construct work, but it seems to me that it has to be "render.render(*args, **kwargs)" to execute the render function in the .py file...._

**Comment by Gingso** [comment]
issue can be closed, i had a typo in the template (special character which was overlooked)

**Reply by robert-hh** [reply]
This topic in the discussions section. It cannot be closed. It just fades away.
